### PR TITLE
Seperate Inactive Cards with Active Cards

### PR DESCRIPTION
**Pull request by aramshiva**
Fixes #4556.

Changes:
- Adds a pill filter to cards page to sort by different things like virtual and physical

Context:
- Before it was messy and unorganized with a jumble of inactive and active cards.

Before: 
image
After:
image

### CODE REVIEW

**Comment by aramshiva** [comment]
For future, if cards in a section are empty then write a message telling the user.

**Review by YodaLightsabr — COMMENTED** [review]
Looks good from my end! I just need to test some cases with different numbers of cards for orgs & users, and then I'll approve.

**Comment by garyhtou** [comment]
I'm chating with Malted about his work on the cards page redesign to see how we can fit this work into his vision! I'll keep you updated

**Comment by aramshiva** [comment]
Ok! Should I close this then for now, because Malted will handle it from here? Or should I keep this open?

**Comment by garyhtou** [comment]
Let's keep this open! I'll update you

**Comment by garyhtou** [comment]
https://github.com/hackclub/hcb/issues/4556

File: app/controllers/events_controller.rb
Comment (garyhtou) [inline app/controllers/events_controller.rb]: Let's create two separate instance variables:
- `@active_stripe_cards`
- `@inactive_stripe_card`

Then, use scopes to filter the two.

**Review by garyhtou — COMMENTED** [review]
Alternative UI: Pill filter with
- All (my cards ordered at the top)
- Active
- Inactive

@YodaLightsabr has some more context on this

File: app/controllers/events_controller.rb
Comment (aramshiva) [reply app/controllers/events_controller.rb]: This actually isnt being used anymore!

File: app/controllers/events_controller.rb
Comment (YodaLightsabr) [inline app/controllers/events_controller.rb]:
```suggestion

```

**Review by YodaLightsabr — DISMISSED** [review]

File: app/controllers/events_controller.rb
Comment (garyhtou) [inline app/controllers/events_controller.rb]: smart!

**Comment by aramshiva** [comment]
Oops! Misclick

File: app/controllers/events_controller.rb
Comment (garyhtou) [inline app/controllers/events_controller.rb]: Rename this param to `status` because that's what we're filtering

Comment (garyhtou) [inline app/controllers/events_controller.rb]:
```suggestion
    @status = %w[virtual physical active inactive].include?(params[:status]) ? params[:status] : nil
```

File: app/controllers/events_controller.rb
Comment (garyhtou) [inline app/controllers/events_controller.rb]:
```suggestion
    all_stripe_cards = case @status
```

File: app/views/events/card_overview.html.erb
Comment (garyhtou) [inline app/views/events/card_overview.html.erb]:
```suggestion
        <%= blankslate "Your teammates don't have any #{@status} cards." %>
```

File: app/views/events/card_overview.html.erb
Comment (garyhtou) [inline app/views/events/card_overview.html.erb]:
```suggestion
        <%= blankslate "This organization doesn't have any #{@status} cards" %>
```

**Review by garyhtou — APPROVED** [review]
🪄 This is good to merge! If you like, figure out how to keep the `view=list` param when switching filters

**Comment by aramshiva** [comment]
I think it should be fine as list is meant to be more general and compacts the information.

**Comment by aramshiva** [comment]
can someone merge? dont have write access.

**Pull request merged**
